src/box/index_def.c

```diff
@@ -113,6 +114,23 @@ index_opts_parse_covered_fields(const char **data, void *opts,
 	return 0;
 }
 
+/** Parse index layout option given as MsgPack in `data' into `opts'. */
+static int
+index_opts_parse_layout(const char **data, void *opts, struct region *region)
+{
+	(void)region;
+	struct index_opts *index_opts = (struct index_opts *)opts;
+	if (mp_typeof(**data) != MP_STR) {
+		diag_set(IllegalParams, "'layout' must be string");
+		return -1;
+	}
+	uint32_t len = 0;
+	const char *str = mp_decode_str(data, &len);
+	if (len > 0)
+		index_opts->layout = xstrndup(str, len);
```

Comment:
It is expected that memory allocation in custom parse will be on `region`. Otherwise memory allocated may leak.

Reply:
Done, however it required an additional `xstrdup()` in `index_opts_normalize()` to avoid `free()` on region-allocated memory in `index_opts_destroy()`.